### PR TITLE
ci: check re-generation of contracts matches checked-in version

### PR DESCRIPTION
**Pull request by huitseeker**

### CODE REVIEW

**Comment by mpenciak** [comment]
I like this CI addition! The only concern I have is whether `forge fmt` is deterministic, and we may get some false positives? 

It seems like everything is Ok for now though, so it may be worth merging and worrying about it later if it comes up.

**Review by mpenciak — APPROVED** [review]

File: .github/workflows/contract_regen.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Run forge fmt
+        run: |
+          forge fmt
+
+      - name: Check discrepancies
+        run: |
+          if [[ `git status --porcelain` ]]; then
+            echo "There are discrepancies between the repository and the source. Please ensure the contracts are up to date."
+            exit 1
+          fi
```
Comment (storojs72) [inline .github/workflows/contract_regen.yml:39]: Minor: newline?

**Review by storojs72 — COMMENTED** [review]
This CI enhancement is implemented as a separate job, which is nice. However, from current [build](https://github.com/lurk-lab/solidity-verifier/actions/runs/5379429232/jobs/9760639224) it is not clear, whether tests will pass with either existing or re-generated data contracts (the original test flow was not invoked). Shouldn't we integrate these two flows into the common sequential one, where unit-tests are executed using re-generated contracts iff there are no discrepancies?

**Review by storojs72 — APPROVED** [review]
Thanks, @huitseeker. Now, It looks more natural from my perspective

**Pull request merged**